packages/headless/doc-parser/src/interface-resolver.ts

```diff
@@ -62,12 +62,36 @@ function filterOverridesAndCombine(
   return members.concat(filtered);
 }
 
+function sortEntities(entities: AnyEntity[]) {
+  const kindOrder: Record<EntityKind, number> = {
+    primitive: 0,
+    'primitive-with-type-alias': 0,
+    object: 0,
+    function: 1,
+  };
+
+  return entities.sort((a, b) => {
+    // Sort by kind (primitives & objects, then functions)
+    let order = kindOrder[a.kind] - kindOrder[b.kind];
+
+    // Sort by optional (mandatory then optional)
+    if (a.kind !== 'function' && b.kind !== 'function') {
+      order += ((a.isOptional ? 1 : 0) - (b.isOptional ? 1 : 0)) / 2;
+    }
+
+    // Sort alphabetically
+    order += a.name.localeCompare(b.name) / 4;
```

Comment:
I'm not clear on why we need to divide by `2` and divide by `4`. Couldn't we just create two partitions, optional and required, sort each and concat?

```
const requiredEntities = nonFuncEntities.filter(entity => !entity.isOptional)
const optionalEntities = nonFuncEntities.filter(entity => entity.isOptional)

// sort & concat
```
The current approach is more efficient, but at the same time, these arrays should not be more than ~10 members long, so I don't think there will be a measurable difference in performance.

Reply:
The division by 2 and 4 were to establish priority (+/- 0.25 will never tip the scale against +/- 0.5 or +/- 1).

I'll change it to use multiple functions instead.